Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4015,6 +4018,110 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
     return cell;
 }
 
+#pragma mark - swipe to show message details
+
+- (void)didPanWithGestureRecognizer:(UIPanGestureRecognizer *)gestureRecognizer
+                           viewItem:(ConversationViewItem *)conversationItem
+{
+    self.currentShowMessageDetailsPanGesture = gestureRecognizer;
+
+    const CGFloat leftTranslation = -1 * [gestureRecognizer translationInView:self.view].x;
```

Comment:
Do we need to clamp this?  Could users reverse the gesture and pan back past the original arrangement?

Reply:
Not that I could reproduce, but I've now clamped it between 0-1.0 anyway.